RFS/src/main/java/com/rfs/common/DocumentReindexer.java

```diff
@@ -1,47 +1,36 @@
 package com.rfs.common;
 
 import java.time.Duration;
-import java.util.Base64;
 import java.util.List;
 
-import io.netty.buffer.Unpooled;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.lucene.document.Document;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
-import reactor.netty.http.client.HttpClient;
 import reactor.util.retry.Retry;
 
 
 public class DocumentReindexer {
     private static final Logger logger = LogManager.getLogger(DocumentReindexer.class);
     private static final int MAX_BATCH_SIZE = 1000; // Arbitrarily chosen
 
-    public static void reindex(String indexName, Flux<Document> documentStream, ConnectionDetails targetConnection) throws Exception {
-        String targetUrl = "/" + indexName + "/_bulk";
-        HttpClient client = HttpClient.create()
-            .host(targetConnection.hostName)
-            .port(targetConnection.port)
-            .headers(h -> {
-                h.set("Content-Type", "application/json");
-                if (targetConnection.authType == ConnectionDetails.AuthType.BASIC) {
-                    String credentials = targetConnection.username + ":" + targetConnection.password;
-                    String encodedCredentials = Base64.getEncoder().encodeToString(credentials.getBytes());
-                    h.set("Authorization", "Basic " + encodedCredentials);
-                }
-            });
+    public static Mono<Void> reindex(String indexName, Flux<Document> documentStream, ConnectionDetails targetConnection) throws Exception {   
+        OpenSearchClient client = new OpenSearchClient(targetConnection);
 
-        documentStream
+        return documentStream
             .map(DocumentReindexer::convertDocumentToBulkSection)  // Convert each Document to part of a bulk operation
             .buffer(MAX_BATCH_SIZE) // Collect until you hit the batch size
+            .doOnNext(bulk -> logger.info(bulk.size() + " documents in current bulk request"))
             .map(DocumentReindexer::convertToBulkRequestBody)  // Assemble the bulk request body from the parts
-            .flatMap(bulkJson -> sendBulkRequest(client, targetUrl, bulkJson)) // Send the request
+            .flatMap(bulkJson -> client.sendBulkRequest(indexName, bulkJson) // Send the request
+                .doOnSuccess(unused -> logger.debug("Batch succeeded"))
+                .doOnError(error -> logger.error("Batch failed", error))
```

Comment:
You'll want to include some identifier so that you can determine the cause and impact of the failure.

Reply:
In this case, the error message includes the response body of the failed POST.  I think that should be sufficient?

Reply:
Oh, I missed the error, sorry.  If error is an exception, you could also do `logger.atError(()->"Label").setCause(error).log`

Reply:
I'll take another pass at all the logging stuff a bit later, if you don't mind; want to get the core functionality in place right now.